File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    
    def dfs(number, total):
        nonlocal answer
        
        if number == len(numbers):
            if total == target: answer += 1
            return
        
        dfs(number + 1, total + numbers[number])
        dfs(number + 1, total - numbers[number])
        
    dfs(0, 0)

    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let result = 0
    
    dfs(0, 0)
    
    function dfs(idx, sum) {
        if (idx === numbers.length) {
            if (sum === target) result += 1
            return
        }
        dfs(idx + 1, sum + numbers[idx])
        dfs(idx + 1, sum - numbers[idx])
    }
    
    return result
}